src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,117 @@
+            break;
+        case "done":
+            if (parseArray.length == 1) {
+                throw new EmptyTaskException();
+            } else {
+                String time = parseArray[1];
+                int index = Integer.parseInt(time) - 1;
+                command = new DoneCommand(index);
+            }
+            break;
+        case "delete":
+            if (parseArray.length == 1) {
+                throw new EmptyTaskException();
+            } else {
+                String rest = parseArray[1];
+                int index = Integer.parseInt(rest) - 1;
+                command = new DeleteCommand(index);
+            }
+            break;
+        case "todo":
+            // Fallthrough
+        case "deadline":
+            // Fallthrough
+        case "event":
+            if (parseArray.length == 1) {
+                throw new EmptyTaskException();
+            } else {
+                String rest = parseArray[1];
+                switch(type) {
+                case "todo":
+                    command = new AddCommand(type, rest, null);
+                break;
+                case "deadline":
+                    if (rest.split("/").length == 1) {
+                        throw new EmptyTimeException("Please specify deadline using \"/by\". (´∀`)");
+                    } else {
+                        String description = rest.split(" /")[0];
+                        try {
+                            String time = rest.split(" /")[1].split(" ", 2)[1];
+                            LocalDate date = LocalDate.parse(time);
+                            command = new AddCommand(type, description, date);
+                        } catch (ArrayIndexOutOfBoundsException ex) {
+                            throw new EmptyTimeException("Please don't leave the deadline blank~ (´∀`)");
+                        } catch (DateTimeParseException ex) {
+                            throw new WrongDateFormatException();
+                        }
+                    }
+                    break;
+                case "event":
+                    if (rest.split("/").length == 1) {
+                        throw new EmptyTimeException("Please specify event time using \"/at\". (´∀`)");
+                    } else {
+                        String description = rest.split(" /")[0];
+                        try {
+                            String time = rest.split(" /")[1].split(" ", 2)[1];
+                            LocalDate date = LocalDate.parse(time);
+                            command =  new AddCommand(type, description, date);
+                        } catch (ArrayIndexOutOfBoundsException ex) {
+                            throw new EmptyTimeException("Please don't leave the event time blank~ (´∀`)");
+                        } catch (DateTimeParseException ex) {
+                            throw new WrongDateFormatException();
+                        }
+                    }
+                }
+            break;
```

Comment:
Perhaps this block of code could be abstracted into multiple method definitions within this class so as to shorten the method length?